bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/discovery/ThingDiscoveryService.java

```diff
@@ -0,0 +1,225 @@
+//            new AbstractMap.SimpleEntry<>("HUE_LIGHT_ROOM_CONTROL", BoschSHCBindingConstants.)
+            );
+    // @formatter:on
+
+    public ThingDiscoveryService() {
+        super(SUPPORTED_THING_TYPES, SEARCH_TIME);
+    }
+
+    @Override
+    public void activate() {
+        logger.trace("activate");
+        final BridgeHandler handler = shcBridgeHandler;
+        if (handler != null) {
+            handler.registerDiscoveryListener(this);
+        }
+    }
+
+    @Override
+    public void deactivate() {
+        logger.trace("deactivate");
+        final BridgeHandler handler = shcBridgeHandler;
+        if (handler != null) {
+            removeOlderResults(new Date().getTime(), handler.getThing().getUID());
+            handler.unregisterDiscoveryListener();
+        }
+
+        super.deactivate();
+    }
+
+    @Override
+    protected void startScan() {
+        if (shcBridgeHandler == null) {
+            logger.debug("The shcBridgeHandler is empty, no manual scan is currently possible");
+            return;
+        }
+
+        doScan();
+    }
+
+    @Override
+    protected synchronized void stopScan() {
+        logger.debug("Stop manual scan on bridge {}",
+                shcBridgeHandler != null ? shcBridgeHandler.getThing().getUID() : "?");
+        super.stopScan();
+        final BridgeHandler handler = shcBridgeHandler;
+        if (handler != null) {
+            removeOlderResults(getTimestampOfLastScan(), handler.getThing().getUID());
+        }
+    }
+
+    @Override
+    public void setThingHandler(@Nullable ThingHandler handler) {
+        if (handler instanceof BridgeHandler) {
+            logger.trace("Set bridge handler {}", handler);
+            shcBridgeHandler = (BridgeHandler) handler;
+        }
+    }
+
+    @Override
+    public @Nullable ThingHandler getThingHandler() {
+        return shcBridgeHandler;
+    }
+
+    public void doScan() {
+        logger.debug("Start manual scan on bridge {}", shcBridgeHandler.getThing().getUID());
+        // use shcBridgeHandler to getDevices()
+        try {
+            List<Room> rooms = shcBridgeHandler.getRooms();
+            logger.debug("SHC has {} rooms", rooms.size());
+            List<Device> devices = shcBridgeHandler.getDevices();
+            logger.debug("SHC has {} devices", devices.size());
+
+            // Write found devices into openhab.log to support manual configuration
+            for (Device d : devices) {
+                logger.info("Found device: name={} id={}", d.name, d.id);
+                if (d.deviceServiceIds != null) {
+                    for (String s : d.deviceServiceIds) {
+                        logger.info(".... service: {}", s);
+                    }
+                }
+            }
+
+            addDevices(devices, rooms);
+        } catch (InterruptedException e) {
+            logger.trace("scan was interrupted");
```

Comment:
The interrput flag should be restored using `Thread.currentThread().interrupt();`, see https://rules.sonarsource.com/java/RSPEC-2142